### ISSUE TITLE
Bug: Where clause with Postgres JSONB nested field

### ISSUE THREAD

**Issue by sayanarijit** [piccolo-orm/piccolo#1103]
Using nested jsonb field in `where` clause is producing unexpected result.

## How to reproduce:

```python
from piccolo.table import Table, create_db_tables_sync
from piccolo import columns as col

class Foo(Table):
    meta = col.JSONB()

def main():
    # Create the table
    create_db_tables_sync(Foo, if_not_exists=True)

    # Create a new objects
    Foo(meta={"a": True, "b": True}).save().run_sync()
    Foo(meta={"a": True, "b": False}).save().run_sync()
    Foo(meta={"a": False, "b": True}).save().run_sync()
    Foo(meta={"a": False, "b": False}).save().run_sync()

    # Query the table
    faulty_results = Foo.select().where(Foo.meta.arrow("a") != Foo.meta.arrow("b")).run_sync()
    for row in faulty_results:
        print(row)

if __name__ == "__main__":
    main()
```

## Expected behavior

```python
{'id': 2, 'meta': '{"a": true, "b": false}'}
{'id': 3, 'meta': '{"a": false, "b": true}'}
```

## Actual behavior

```python
{'id': 1, 'meta': '{"a": true, "b": true}'}
{'id': 2, 'meta': '{"a": true, "b": false}'}
{'id': 3, 'meta': '{"a": false, "b": true}'}
{'id': 4, 'meta': '{"a": false, "b": false}'}
```

Repository for convenience: https://github.com/sayanarijit/piccolobug

**Comment by sinisaos** [comment]
@sayanarijit I don't know if this is a bug or not, but we can achieve the expected behavior in three ways.
1. using complex checking for differences between `meta.a` and `meta.b`
```python
faulty_results = (
      Foo.select()
      .where(
          (Foo.meta.arrow("a") == True) & (Foo.meta.arrow("b") == False)
          | (Foo.meta.arrow("a") == False) & (Foo.meta.arrow("b") == True)
      )
      .run_sync()
  )
```
2. using raw sql
```python
faulty_results = Foo.raw(
    "SELECT * FROM foo WHERE meta->'a' != meta->'b'"
).run_sync()
```
3. using [WhereRaw](https://piccolo-orm.readthedocs.io/en/latest/piccolo/query_clauses/where.html#whereraw) to combine Piccolo query with raw sql
```python
faulty_results = Foo.select().where(WhereRaw("meta->'a' != meta->'b'")).run_sync()
```

**Comment by sayanarijit** [comment]
Thanks @sinisaos for the alternate options. However, I think it’s intuitive to assume Piccolo’s arrow behaviour in where clause should match the behaviour in other ORMs like peewee and sqlalchemy (Foo.meta[“a”] != Foo.meta[“b”], else it comes off as an unexpected surprise.

**Comment by sinisaos** [comment]
@sayanarijit The query is not good because if we print a querystring for  
```python
print(Foo.select().where(Foo.meta.arrow("a") != Foo.meta.arrow("b")))
```
the result is this
```bash
SELECT ALL "foo"."id" AS "id", "foo"."meta" AS "meta" FROM "foo" WHERE 
"foo"."meta" -> 'a' != "foo"."meta" # missing -> 'b'
```
Missing `meta.b` value. @dantownsend, can you check it out when you have time?

**Comment by dantownsend** [comment]
@sinisaos Yeah, the way it was originally implemented doesn't allow comparing one arrow function to another.

A few months ago I made `QueryString` more powerful (you can now do stuff like `some_querystring == some_other_querystring`).

Here's a PR showing what I mean:

https://github.com/piccolo-orm/piccolo/pull/1112

Unfortunately a couple of tests are failing. Once we figure out why those tests are failing, I can release it.

**Comment by dantownsend** [comment]
This should be fixed in `piccolo==1.22.0`.
